core/src/main/java/com/linecorp/armeria/common/http/PathParamExtractor.java

```diff
@@ -72,11 +77,11 @@ static DynamicPath of(String path) {
     private final String skeleton;
 
     /**
-     * Set of the variables declared in this dynamic path.
+     * Set of the variables declared in this path param extractor.
      */
     private final ImmutableSet<String> variables;
 
-    DynamicPath(String path) {
+    private PathParamExtractor(String path) {
```

Comment:
I wonder if we could let users just use the public constructor given that `of()` just delegates to this constructor. i.e. How about removing `of()` and making this constructor public?